include/knowhere/config.h

```diff
@@ -494,13 +494,29 @@ class BaseConfig : public Config {
     CFG_STRING metric_type;
     CFG_INT k;
     CFG_INT num_build_thread;
+    CFG_BOOL retrieve_friendly;
+    CFG_STRING data_path;
+    CFG_STRING index_prefix;
     CFG_FLOAT radius;
     CFG_FLOAT range_filter;
     CFG_BOOL trace_visit;
     CFG_BOOL enable_mmap;
     CFG_BOOL for_tuning;
     KNOHWERE_DECLARE_CONFIG(BaseConfig) {
         KNOWHERE_CONFIG_DECLARE_FIELD(metric_type).set_default("L2").description("metric type").for_train_and_search();
+        KNOWHERE_CONFIG_DECLARE_FIELD(retrieve_friendly)
+            .description("index should hold raw data for fast retrieve")
+            .set_default(false)
+            .for_train();
+        KNOWHERE_CONFIG_DECLARE_FIELD(data_path)
+            .description("raw data path.")
+            .allow_empty_without_default()
+            .for_train();
+        KNOWHERE_CONFIG_DECLARE_FIELD(index_prefix)
+            .description("path to load or save index.")
+            .allow_empty_without_default()
+            .for_train()
+            .for_deserialize();
```

Comment:
Why we put this into the highest layer of Knowhere Config?

Reply:
Some indexes may load data from/serialize binary to disk files.